src/symmetric.jl

```diff
@@ -987,6 +988,7 @@ See also [`hermitianpart!`](@ref) for the corresponding in-place operation.
     This function requires Julia 1.10 or later.
 """
 hermitianpart(A::AbstractMatrix, uplo::Symbol=:U) = Hermitian(_hermitianpart(A), uplo)
+hermitianpart(x::Number) = real(x)
```

Comment:
Should this perhaps be

```suggestion
hermitianpart(x::Number) = float(real(x))
```

to mimick the `(x + x')/2` behavior of arrays

```julia
julia> hermitianpart([5;;])
1×1 Hermitian{Float64, Matrix{Float64}}:
 5.0
```

?

Reply:
I'm not sure I agree.  Why does it need to match the type returned for arrays?

Reply:
And even for arrays, the type is not always floating-point:
```jl
julia> hermitianpart([5//1;;])
1×1 Hermitian{Rational{Int64}, Matrix{Rational{Int64}}}:
 5
```

Reply:
It was more like an open question. The association that came to my mind is the norm of numbers, which we also promote to floating-point types, perhaps to mimick the `sqrt(x^2)` computation, without actually computing it.

Reply:
The `norm` of a `Complex` number has to be a floating-point type regardless.  And `norm(::Integer)` also needs to be floating-point to avoid spurious overflow for fixed-width integers.  Similarly for `norm(::Rational)`.   So `norm` pretty much has to be floating point even for scalars.

This isn't the case for `hermitianpart`, since `real(x)` should always work without conversion.